src/openrct2/windows/footpath_scenery.c

```diff
@@ -0,0 +1,484 @@
+{
+    switch (widgetIndex)
+    {
+    case WIDX_CLOSE:
+        window_close(w);
+        break;
+    case WIDX_CLEAR_RULES:
+        window_footpath_scenery_clear_rules(w);
+        break;
+    case WIDX_ADD:
+        window_footpath_scenery_add(w);
+        break;
+    }
+}
+
+static void window_footpath_scenery_mousedown(int widgetIndex, rct_window * w, rct_widget * widget)
+{
+    switch (widgetIndex)
+    {
+    case WIDX_OBJECT_DROPDOWN:
+        window_footpath_scenery_path_scenery_dialog(w, widget - 1, widgetIndex);
+        break;
+    }
+}
+
+static void window_footpath_scenery_dropdown(rct_window * w, int widgetIndex, int dropdownIndex)
+{
+    if (widgetIndex == WIDX_OBJECT_DROPDOWN)
+    {
+        window_footpath_scenery_object_dropdown(w, widgetIndex, dropdownIndex);
+    }
+}
+
+static void window_footpath_scenery_update(rct_window * w)
+{
+    window_invalidate(w);
+}
+
+static void window_footpath_scenery_invalidate(rct_window * w)
+{
+    colour_scheme_update(w);
+
+    if (_footpath_scenery_selected_sceneryEntry != NULL)
+    {
+        set_format_arg(0, rct_string_id, _footpath_scenery_selected_sceneryEntry->name);
+    }
+    else
+    {
+        set_format_arg(0, rct_string_id, STR_PATH_SCENERY_NOTHING);
+    }
+}
+
+static void window_footpath_scenery_paint(rct_window * w, rct_drawpixelinfo * dpi)
+{
+    window_draw_widgets(w, dpi);
+
+    gfx_draw_string_left(dpi, STR_PATH_SCENERY_ON_PATH, NULL, COLOUR_BLACK, w->x + PADDING, w->y + YPL(10) + 3);
+}
+
+static void window_footpath_scenery_path_scenery_dialog(rct_window * w, rct_widget * widget, int widgetIndex)
+{
+    int numSceneryTypes     = 1;
+    gDropdownItemsFormat[0] = -1;
+    gDropdownItemsArgs[0]   = SPR_NONE;
+
+    for (uint16 sceneryId = 0x100; sceneryId < 0x10F; sceneryId++)
+    {
+        int pathBitIndex = sceneryId - 0x100;
+
+        if (get_footpath_item_entry(pathBitIndex) == (rct_scenery_entry *)-1)
+            continue;
+
+        if (!scenery_is_invented(sceneryId))
+            continue;
+
+        rct_scenery_entry * sceneryEntry = get_footpath_item_entry(pathBitIndex);
+
+        gDropdownItemsFormat[numSceneryTypes] = -1;
+        gDropdownItemsArgs[numSceneryTypes]   = sceneryEntry->image;
+        numSceneryTypes++;
+    }
+
+    window_dropdown_show_image(w->x + widget->left, w->y + widget->top, widget->bottom - widget->top + 1, w->colours[1], 0,
+                               numSceneryTypes, 50, 50, gAppropriateImageDropdownItemsPerRow[numSceneryTypes]);
+    return;
+}
+
+static void window_footpath_scenery_object_dropdown(rct_window * w, int widgetIndex, int dropdownIndex)
+{
+    int numSceneryTypes = 0;
+    if (dropdownIndex == 0)
+    {
+        _footpath_scenery_selected_sceneryEntry = NULL;
+        _footpath_scenery_selected_sceneryId    = 0;
+    }
+    else if (dropdownIndex != -1)
+    {
+        dropdownIndex--;
+        uint16 sceneryId;
+
+        for (sceneryId = 0x100; sceneryId < 0x10F; sceneryId++)
```

Comment:
We're aiming to reduce the number of magic numbers. Could you add defines for these? Even if only for this file.